Stack/Opc.Ua.Core/Stack/Server/ServerBase.cs

```diff
@@ -805,6 +805,27 @@ protected virtual void OnCertificateUpdate(object sender, CertificateUpdateEvent
                 }
             }
 
+            foreach (EndpointDescription endpointDescription in m_endpoints.Where(endpoint => RequireEncryption(endpoint)))
```

Comment:
this loop or its content to build the cert chain looks familiar. Maybe a helper sould be sueful to avoid duplicat code.